pkg/cluster/workloadidentityresources.go

```diff
@@ -124,3 +130,37 @@ func (m *manager) generateAuthenticationConfig() (*configv1.Authentication, erro
 		},
 	}, nil
 }
+
+// getPlatformWorkloadIdentityFederatedCredName returns a federated identity credential name.
+// name length 3-120 characters. It must be alphanumeric, dash, underscore.
+func (m *manager) getPlatformWorkloadIdentityFederatedCredName(sa string, identity api.PlatformWorkloadIdentity) (string, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return "", fmt.Errorf("getPlatformWorkloadIdentityFederatedCredName called for a CSP cluster")
+	}
+
+	if strings.TrimSpace(sa) == "" {
+		return "", fmt.Errorf("service account name is required")
+	}
+
+	identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+	if err != nil {
+		return "", err
+	}
+	clusterResourceId, err := azure.ParseResourceID(m.doc.OpenShiftCluster.ID)
+	if err != nil {
+		return "", err
+	}
+
+	clusterResourceKey := fmt.Sprintf("%s-%s-%s", clusterResourceId.SubscriptionID, clusterResourceId.ResourceGroup, clusterResourceId.ResourceName)
+	name := fmt.Sprintf("%s-%s-%s", clusterResourceKey, sa, identityResourceId.ResourceName)
+	// the base-36 encoded string of a SHA-224 hash will typically be around 43 to 44 characters long.
+	hash := sha256.Sum224([]byte(name))
+	encodedName := (&big.Int{}).SetBytes(hash[:]).Text(36)
```

Comment:
Nit: please use constants for hardcoded numbers

Reply:
updated